src/SourceBuild/content/prep.sh

```diff
@@ -67,57 +68,76 @@ done
 
 # Attempting to bootstrap without an SDK will fail. So either the --no-sdk flag must be passed
 # or a pre-existing .dotnet SDK directory must exist.
-if [[ "$buildBootstrap" == "true" && "$installDotnet" == "false" && ! -d $SCRIPT_ROOT/.dotnet ]]; then
+if [ "$buildBootstrap" == true ] && [ "$installDotnet" == false ] && [ ! -d "$SCRIPT_ROOT/.dotnet" ]; then
     echo "  ERROR: --no-sdk requires --no-bootstrap or a pre-existing .dotnet SDK directory.  Exiting..."
-    exit -1
+    exit 1
 fi
 
 # Check to make sure curl exists to download the archive files
 if ! command -v curl &> /dev/null
 then
     echo "  ERROR: curl not found.  Exiting..."
-    exit -1
+    exit 1
+fi
+
+GIT_DIR="$SCRIPT_ROOT/.git"
+
+if [ -f "$GIT_DIR/config" ]; then
+  if [ -n "$sourceUrl" ] || [ -n "$sourceVersion" ]; then
+    echo "ERROR: $SCRIPT_ROOT is a git repository, --source-repository and --source-version cannot be used."
+    exit 1
+  fi
+else
+  if [ -z "$sourceUrl" ] || [ -z "$sourceVersion" ]; then
+    echo "ERROR: $SCRIPT_ROOT is not a git repository, --source-repository and --source-version must be specified."
+    exit 1
+  fi
+
+  # We need to add "fake" .git/ files when not building from a git repository
+  mkdir -p "$GIT_DIR"
```

Comment:
Is there no support for source-link to read this info from anyplace besides git metadata?

Reply:
As noted below, more than sourcelink depend on this and more can come in the future